src/marqo/marqo_cloud_instance_mappings.py

```diff
@@ -25,21 +26,27 @@ def get_control_base_url(self) -> str:
         return f"{self._control_base_url}/api"
 
     def get_index_base_url(self, index_name: str) -> str:
+        """Returns the index_name's base URL regardless of its status.
+
+        Raises:
+            MarqoCloudIndexNotFoundError: if index_name is not found in any status.
+        """
         self._refresh_urls_if_needed(index_name)
-        if index_name in self._urls_mapping[IndexStatus.READY]:
-            return self._urls_mapping[IndexStatus.READY][index_name]
-        if index_name in self._urls_mapping[IndexStatus.CREATING]:
-            raise MarqoCloudIndexNotReadyError(index_name)
+
+        for cloud_status, indexes in self._urls_mapping.items():
+            if index_name in indexes:
+                return indexes[index_name]
```

Comment:
This is going to miss `MarqoCloudIndexNotReadyError` (index creating). Also that our tests aren't catching this

Reply:
This method's contract has shifted in this PR. The idea is to simplify the `get_index_base_url()` logic. We let users' context that an index is creating (from the console, for example) to interpret any 500 error they receive when they prematurely try to use the index.

Reply:
The relevant test is here: https://github.com/marqo-ai/py-marqo/blob/pandu/no-warnings-on-index-instantiation/tests/v0_tests/test_marqo_cloud_instance_mapping.py#L174